tests/integration/declarative/declarative_test.go

```diff
@@ -1,384 +0,0 @@
-package declarative_test
```

Comment:
To reviewer:

The reason to remove this declarative test:
1. I explicitly cast the generic object to manifest in [here](https://github.com/kyma-project/lifecycle-manager/pull/1336/files#diff-45632cfa5b5d235890e597e29448382f7c61fc85a55d1629de012328056e69a2R496) which made this declarative test not working anymore, since it use a customized TestAPI to test the reconciler function.
2. Since we decided to completely [remove this declarative library](https://github.com/kyma-project/lifecycle-manager/issues/1006), there is no need to support such tests anymore.
3. We have integration test for the manifest controller which tests against manifest CR specifically.

Reply:
Agreed 👌